src/main/scala/is/hail/utils/TextTableReader.scala

```diff
@@ -205,38 +207,53 @@ object TextTableReader {
 
     val sb = new StringBuilder
 
+    val useColIndices = if (!useCols.isEmpty) {
+      columns.zipWithIndex.flatMap { case (str,index) => if (useCols.contains(str)) Some(index) else None }
```

Comment:
If useCols is an `Option[Set[String]]`, then this can be

```
useCols.map(s => 
  columns.zipWithIndex.flatMap { case (str, index) => if (useCols.contains(str)) Some(index) else None)
  .getOrElse(columns.indices.toArray)
```